timesketch/lib/analyzers/sequence_sessionizer.py

```diff
@@ -0,0 +1,158 @@
+            self.process_event(first_event)
+
+            for event in events:
+                curr_timestamp = event.source.get('timestamp')
+                if curr_timestamp - last_timestamp > self.max_time_diff_micros:
+                    self.flush_events(drop=True)
+
+                self.process_event(event)
+                last_timestamp = curr_timestamp
+        except StopIteration:
+            pass
+
+        self.query = self.get_query_string()
+
+        self.sketch.add_view('Session view',
+                             self.NAME,
+                             query_string=self.query)
+
+        return ('Sessionizing completed, number of {0:s} session created:'
+                ' {1:d}'.format(self.session_type, self.session_num))
+
+    def annotateEvent(self, event, session_num):
+        """Add an session_type attribute with a session_num to event.
+
+        Args:
+            event: Event to annotate.
+            session_num: Session number for the event.
+        """
+        event.add_attributes({self.session_type: session_num})
+        event.commit()
+
+    def process_event(self, event):
+        """Process event depending on if the event is significant for the
+        searched event sequence.
+
+        Event is significant if it matches the current
+        event in the event_seq or if in the timeline it is between two
+        consistent events from the event_seq.
+
+        Args:
+            event: Event to process.
+        """
+        if self.match_event(event):
+            self.recording = True
+            self.event_storage.append(event)
+            self.num_event_to_find += 1
+
+            if self.num_event_to_find == len(self.event_seq):
+                self.flush_events()
+            return
+
+        if self.recording:
+            self.event_storage.append(event)
+
+    def flush_events(self, drop=False):
+        """Annotates or clears events in event_storage according to the flag
+        drop and resets session recording state.
+
+        Args:
+            drop: Indicator if the stored events should not be commited.
+        """
+        if not drop:
+            self.session_num += 1
+            for event in self.event_storage:
+                self.annotateEvent(event, self.session_num)
+
+        self.recording = False
+        self.num_event_to_find = 0
+        self.event_storage = []
+
+    def match_event(self, event):
+        """Compare event with the event to search for in the event sequence.
+
+        Args:
+            event: Event to compare with the one in the event sequence.
+
+        Returns:
+            If event matches the event to search for in the event
+        sequence.
+        """
+        event_to_match = self.event_seq[self.num_event_to_find]
+
+        for key, value in event_to_match.items():
+            if value != event.source.get(key):
+                return False
+        return True
+
+    def get_query_string(self):
```

Comment:
If this function is meant to be overriden by other classes, please specify it in the docstring.